src/EditorFeatures/Core/Shared/Tagging/AsynchronousTagger.cs

```diff
@@ -65,6 +67,20 @@ internal sealed partial class AsynchronousTagger<TTag> : ITagger<TTag>, IDisposa
             _tagSource.Resumed += OnResumed;
         }
 
+        private TaggerDelay GetBatchThrottleDelay(TaggerDelay uiUpdateDelay)
+        {
+            // If the tagger is looking for near immediate UI updates, then we don't want to batch 
+            // changes for too long, otherwise the end-to-end tag time may be noticible.
+            if (uiUpdateDelay == TaggerDelay.NearImmediate)
+            {
+                return TaggerDelay.NearImmediate;
+            }
+
+            // Otherwise, batching for a quarter second is a nice way to space change notifications 
+            // out, while still appearing quickly to the user.
+            return TaggerDelay.Short;
```

Comment:
That's a fair critique :)

Here are my thoughts on the matter.   Tagging actually has *three* throttles in uses from the point in time where an event is triggered, all the way to the point where the editor is notified.  The throttles are:

1. We actually try to batch up trigger events.  i.e. if a bunch of a events happen quickly, we'll try to batch them and only actually start tagging once a specified TaggerDelay completes.  This delay is specified when you create the ITaggerEventSource.
2. Once tags have been computed on our background thread, we marshal them over to the UI thread, passing in a TaggerDelay for how long we want to want before proceeding to execute on the UI thread.  This is handled by _uiUpdateDelayInMS, and is the piece i now expose in this PR.
3. Once we proceed to execute again on the UI thread we push the changed spans into the 'BatchChangeNotifier'.  This component provides a final throttle so we can try to notify the editor about all changed spans in one go, instead of having a back and forth with them in a small period of time.


I'm honestly not a huge fan of how this all works.  It's not clear to me if the BatchChangeNotifier has a lot of value.  After all, we've already specified some amount of delay to wait after computing tags before updating the UI.   How likely is it that another event came in, we waited for it to be throttled, then we produced new tags, and then forwarded that to the UI thread (with another delay), and all that happened before the batch change notifier had time to update VS?

It seems like it could potentially be removed at this point, and we could stick with just a throttle on the event source side, and delays after computing tags before telling the editor.

What do you think @jasonmalinowski ?